rust/otel-arrow-rust/src/pdata/otlp/primitive_encoders.rs

```diff
@@ -0,0 +1,505 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module contains primitive-field encoding support for protobufs.
+///
+/// - EncodedLen impls for primitive visitors.
+/// - Accumulate helper visitable
+use crate::pdata::otlp::PrecomputedSizes;
+
+/// Wire type constants for protobuf encoding
```

Comment:
nit: it might be worth linking to the documentation here by way of reference
https://protobuf.dev/programming-guides/encoding/